### PR TITLE
feat: use `to_pyarrow` instead of `to_pandas` in the interactive repr

### PR DESCRIPTION
**Pull request by jcrist**
Removes the use of pandas in this codepath, and provides more uniform handling of nested datatypes.

### CODE REVIEW

**Comment by jcrist** [comment]
This needs some more work, moving it to a draft for now.

**Comment by cpcloud** [comment]
Anything I can do to help get this merged?

**Comment by cpcloud** [comment]
Ok, I think this is ready for review now.

**Comment by cpcloud** [comment]
Clouds are passing:


```
…/ibis on  pyarrow-repr is 📦 v7.0.0 via 🐍 v3.10.12 via ❄️  impure (ibis-3.10.12-env)
❯ pytest -m 'bigquery or snowflake' --snapshot-update -n auto --dist loadgroup -q
bringing up nodes...
xx.xx..x..x.....x.xx........x....xx..........xxx........xx......x.x........................x........x...x........x....x..............x..........xx....x.....x.x.........................x..... [  6%]
..................x.............x...x...............x....................s.x......x..xx..............................xxx...x......x....x.x.................................................... [ 13%]
.....................x..................x............x...x........x..x.......x...............................x.........x.............................x...x.....x...x.....................xx... [ 19%]
.........x..x..........x...x....x......xx.x.................x........x....xxx.........x.........................x........xxxx..xxxxxxx...x..x...xxx..x..x....x..x....x..x..................... [ 26%]
.......x.x.............x.................x.......x..x.x....x.xx...............x...x.x.........x....x.x..x...x..............xx.........xx....xx...x..x.....x...x......xx..x....x.............x. [ 32%]
...x...x..x......x...x...xx......xx..x..x..x...x.....x......x.x........x...................x...x..x........x...x..........xx....x...xx..x.........xx........sx............x...x.....x.xxxxxxxx [ 39%]
xxx..xxxx.xx....xx...xx.........x.........x.........x.....x........xx.......xx.xxx.......xx..x.x..x..xx.x...xx..x..xx.xxx..xx..xxxx..x....x...x.x..x.....................x......x..xxx..x..xx. [ 45%]
...x.xx.x...x.x..x..xx..xxx.x....x.xx.x.....x.....x...x.x....xx.xx...x..x.x.sssxsssssssssssssssssss..x...............x........x..x......x..x...xxx.x..x.x....xx..xxxx....x.....x.............. [ 52%]
.........................x..............................................................................................x.................................................x................... [ 58%]
....................................x..x.xx.x..........x.........x..x..........x..x...x..x..x...........x...................x.x.....x.....xx..................x...x........x....x..x..x......x [ 65%]
..xxxxxxxxxxxxxxxx.x..xxx.x.x.xx.x..x...sxx......x.xs...x...xs...x..xxxxx.x...xx.xxxxxxxxxxxxx.xxx.x..x.x.x....xx.....x.....x...x...xx...........x.x.......x..x...x.......xx.....x........xx.. [ 72%]
x.......x....x............x..x..x..............x.x...x..x...........x....x........x......x.x.x..x...........x...x..x..........x..x..xx......xx.x...x.......x...........x.................s.... [ 78%]
......................x.x..................s.................................x.x...........xxxx.xx........................................................xx...........x......x............... [ 85%]
.x......................x.......x...............x...........................................x...x......x....x................................................................................. [ 91%]
................................................................................x............................................................................................................. [ 98%]
.......x.................s.......................                                                                                                                                              [100%]
2427 passed, 30 skipped, 442 xfailed in 445.18s (0:07:25)
```

File: ibis/backends/impala/__init__.py [hunk 1]
```diff
@@ -1378,3 +1380,39 @@ def write_dataframe(
         """
         writer = DataFrameWriter(self, df)
         return writer.write_csv(path)
+
+    def to_pyarrow(
```
Comment (kszucs) [inline ibis/backends/impala/__init__.py:1384]: Wow!

**Review by kszucs — APPROVED** [review]
LGTM!

**Pull request merged**